lib/public/TaskProcessing/Task.php

```diff
@@ -0,0 +1,259 @@
+	/**
+	 * @since 30.0.0
+	 */
+	public const STATUS_SUCCESSFUL = 3;
+	/**
+	 * @since 30.0.0
+	 */
+	public const STATUS_RUNNING = 2;
+	/**
+	 * @since 30.0.0
+	 */
+	public const STATUS_SCHEDULED = 1;
+	/**
+	 * @since 30.0.0
+	 */
+	public const STATUS_UNKNOWN = 0;
+
+	/**
+	 * @psalm-var self::STATUS_*
+	 */
+	protected int $status = self::STATUS_UNKNOWN;
+
+	/**
+	 * @param array<string,mixed> $input
+	 * @param string $appId
+	 * @param string|null $userId
+	 * @param null|string $identifier An arbitrary identifier for this task. max length: 255 chars
+	 * @since 30.0.0
+	 */
+	final public function __construct(
+		protected readonly string $taskTypeId,
+		protected array $input,
+		protected readonly string $appId,
+		protected readonly ?string $userId,
+		protected readonly ?string $identifier = '',
+	) {
+	}
+
+	/**
+	 * @since 30.0.0
+	 */
+	final public function getTaskTypeId(): string {
+		return $this->taskTypeId;
+	}
+
+	/**
+	 * @psalm-return self::STATUS_*
+	 * @since 30.0.0
+	 */
+	final public function getStatus(): int {
+		return $this->status;
+	}
+
+	/**
+	 * @psalm-param self::STATUS_* $status
+	 * @since 30.0.0
+	 */
+	final public function setStatus(int $status): void {
+		$this->status = $status;
+	}
+
+	/**
+	 * @param ?DateTime $at
+	 * @since 30.0.0
+	 */
+	final public function setCompletionExpectedAt(?DateTime $at): void {
+		$this->completionExpectedAt = $at;
+	}
+
+	/**
+	 * @return ?DateTime
+	 * @since 30.0.0
+	 */
+	final public function getCompletionExpectedAt(): ?DateTime {
+		return $this->completionExpectedAt;
+	}
+
+	/**
+	 * @return int|null
+	 * @since 30.0.0
+	 */
+	final public function getId(): ?int {
+		return $this->id;
+	}
+
+	/**
+	 * @param int|null $id
+	 * @since 30.0.0
+	 */
+	final public function setId(?int $id): void {
+		$this->id = $id;
+	}
+
+	/**
+	 * @since 30.0.0
+	 */
+	final public function setOutput(?array $output): void {
+		$this->output = $output;
+	}
+
+	/**
+	 * @return array<array-key, mixed>|null
+	 * @since 30.0.0
+	 */
+	final public function getOutput(): ?array {
+		return $this->output;
+	}
+
+	/**
+	 * @return array<array-key, mixed>
+	 * @since 30.0.0
+	 */
+	final public function getInput(): array {
+		return $this->input;
+	}
+
+	/**
+	 * @return string
+	 * @since 30.0.0
+	 */
+	final public function getAppId(): string {
+		return $this->appId;
+	}
+
+	/**
+	 * @return null|string
+	 * @since 30.0.0
+	 */
+	final public function getIdentifier(): ?string {
+		return $this->identifier;
+	}
+
+	/**
+	 * @return string|null
+	 * @since 30.0.0
+	 */
+	final public function getUserId(): ?string {
+		return $this->userId;
+	}
+
+	/**
+	 * @psalm-return array{id: ?int, status: self::STATUS_*, userId: ?string, appId: string, input: ?array<array-key, mixed>, output: ?array<array-key, mixed>, identifier: ?string, completionExpectedAt: ?int, progress: ?float}
+	 * @since 30.0.0
+	 */
+	public function jsonSerialize(): array {
+		return [
+			'id' => $this->getId(),
```

Comment:
```suggestion
			'id' => $this->getId(),
			'type' => $this->getTaskTypeId(),
```
We need the type here otherwise it's not included when getting a task with `/ocs/v2.php/taskprocessing/task/TASK_ID` for example.

Reply:
Wow. Nice catch!